src/components/tooltip/tooltip.js

```diff
@@ -3,392 +3,405 @@
+        return;
+      }
       setVisible.value = !!value;
 
       if (!setVisible.queued) {
         if (value) {
           setVisible.queued = true;
           showTimeout = $timeout(function() {
-            scope.visible = setVisible.value;
+            scope.mdVisible = setVisible.value;
             setVisible.queued = false;
             showTimeout = null;
-
             if (!scope.visibleWatcher) {
-              onVisibleChanged(scope.visible);
+              onVisibleChanged(scope.mdVisible);
             }
-          }, scope.delay);
+          }, scope.mdDelay);
         } else {
           $mdUtil.nextTick(function() {
-            scope.visible = false;
-            if (!scope.visibleWatcher)
+            scope.mdVisible = false;
+            if (!scope.visibleWatcher) {
               onVisibleChanged(false);
+            }
           });
         }
       }
     }
 
+    function onVisibleChanged(isVisible) {
+      isVisible ? showTooltip() : hideTooltip();
+    }
+
     function showTooltip() {
-      //  Do not show the tooltip if the text is empty.
-      if (!element[0].textContent.trim()) return;
-
-      // Insert the element and position at top left, so we can get the position
-      // and check if we should display it
-      element.css({top: 0, left: 0});
-      tooltipParent.append(element);
-
-      // Check if we should display it or not.
-      // This handles hide-* and show-* along with any user defined css
-      if ( $mdUtil.hasComputedStyle(element, 'display', 'none')) {
-        scope.visible = false;
-        element.detach();
-        return;
+      // Do not show the tooltip if the text is empty.
+      if (!element[0].textContent.trim()) {
+        throw new Error('Text for the tooltip has not been provided. ' +
+            'Please include text within the mdTooltip element.');
       }
 
-      updatePosition();
+      var attachTo = angular.element(document.body);
 
-      $animate.addClass(content, SHOW_CLASS).then(function() {
-        element.addClass(SHOW_CLASS);
-      });
-    }
-
-    function hideTooltip() {
-      $animate.removeClass(content, SHOW_CLASS).then(function(){
-        element.removeClass(SHOW_CLASS);
-        if (!scope.visible) element.detach();
-      });
-    }
+      var panelAnimation = $mdPanel.newPanelAnimation()
+          .openFrom(parent)
+          .closeTo(parent)
+          .withAnimation({
+            open: 'md-show',
+            close: 'md-hide'
+          });
 
-    function updatePosition() {
-      if ( !scope.visible ) return;
+      var panelConfig = {
+        attachTo: attachTo,
+        template: element.html(),
+        propagateContainerEvents: true,
+        panelClass: 'md-tooltip ' + origin,
+        animation: panelAnimation,
+        position: panelPosition,
+        zIndex: scope.mdZIndex,
+        focusOnOpen: false
+      };
 
-      updateContentOrigin();
-      positionTooltip();
+      panelRef = $mdPanel.create(panelConfig);
```

Comment:
This seems very inefficient, because it creates a new panel ref on every open. I think we should create one only after the first open and then re-use it.